src/ssl.c

```diff
@@ -3473,14 +3472,13 @@ static int ProcessBuffer(WOLFSSL_CTX* ctx, const unsigned char* buff,
                     FreeX509(ssl->ourCert);
                     if (ssl->ourCert) {
                         XFREE(ssl->ourCert, ssl->heap, DYNAMIC_TYPE_X509);
+                        ssl->ourCert = NULL;
                     }
                 #endif
             }
             XMEMCPY(&ssl->buffers.certificate, &der, sizeof(der));
             #ifdef OPENSSL_EXTRA
-                ssl->ourCert = wolfSSL_X509_d2i(NULL,
-                                              ssl->buffers.certificate->buffer,
-                                              ssl->buffers.certificate->length);
+                ssl->keepCert = 1; /* hold cert for ssl lifetime */
```

Comment:
So every user now has to "pay" for this on every connection?  Keeping memory and resources for something that 99% probably will never use.  I'm having a lot of trouble justifying that.